showcase/app/routes/page-components/advanced-table.ts

```diff
@@ -24,149 +27,30 @@ export const clone = <T>(arr: T[]): T[] => {
-    | string;
-  service?:
-    | {
-        text: string;
-        colspan?: number;
-      }
-    | string;
-  description?: string;
-  email?: string;
-}
-
-interface Policy {
-  id: number;
-  name: string;
-  description: string;
-  status: string;
-  isOpen?: boolean;
-  children?: Policy[];
-}
-
-interface PolicyCustom {
-  id: number;
-  name: string;
-  description: string;
-  status: string;
-  isOpen?: boolean;
-  data?: Policy[];
-}
-
 export default class PageComponentsAdvancedTableRoute extends Route {
-  async model() {
-    const responseMusic = await fetch('/api/folk.json');
-    const responseUserData = await fetch('/api/mock-users.json');
-    const responseClusters = await fetch('/api/mock-clusters-with-status.json');
-    const responseSelectableData = await fetch(
-      '/api/mock-selectable-data.json',
-    );
-    const responseManyColumns = await fetch('/api/mock-many-columns.json');
-    const responseSpanningManual = await fetch(
-      '/api/mock-spanning-cells-manual.json',
-    );
-    const responseNested = await fetch('/api/mock-nested-rows.json');
-    const responseNestedCustom = await fetch(
-      '/api/mock-nested-rows-custom.json',
-    );
-
-    const { data: music } = (await responseMusic.json()) as Record<
-      'data',
-      MusicEntity[]
-    >;
-    const userData = (await responseUserData.json()) as User[];
-    const clusters = (await responseClusters.json()) as Cluster[];
-    const selectableData =
-      (await responseSelectableData.json()) as SelectableItem[];
-    const manyColumns =
-      (await responseManyColumns.json()) as UserWithMoreColumns[];
-    const spanningManualData =
-      (await responseSpanningManual.json()) as SpanningEntity[];
-    const nestedData = (await responseNested.json()) as Policy[];
-    const nestedDataCustom =
-      (await responseNestedCustom.json()) as PolicyCustom[];
-
+  model() {
     return {
-      music: music.map((record) => ({ id: record.id, ...record.attributes })),
+      music: folkMusicData,
       userDataShort: clone(userData.slice(0, 5)),
-      clusters,
-      spanningManualData,
-      selectableData,
-      selectableDataDemo1: clone(selectableData),
-      selectableDataDemo2: clone(selectableData),
-      userData,
+      clusters: clusterData,
+      spanningManualData: spanningCellData,
+      selectableData: selectableItemsData,
+      selectableDataDemo1: clone(selectableItemsData),
+      selectableDataDemo2: clone(selectableItemsData),
+      userData: userData,
       userDataDemo3: clone(userData.slice(0, 16)),
       userDataDemo4: clone(
         userData.slice(0, 4).map((user) => ({ ...user, isAnimated: false })),
       ),
-      manyColumns,
-      nestedData,
-      nestedDataCustom,
+      manyColumns: userWithManyColumnsData,
+      nestedData: policyData,
+      nestedDataCustom: policyData.map((policy) => {
+        const { children, ...rest } = policy;
+        return {
+          ...rest,
+          data: children,
```

Comment:
[question] do you know if we modify any of the `children` or `data` items? otherwise we have to deep clone it (it's an array of objects)

Reply:
The example in the showcase seems to be working just fine, the data is never modified/updated after this.